cli/cmd/initnodes.go

```diff
@@ -398,3 +423,96 @@ func gethInit(ctx context.Context, cfg InitConfig, dir string) error {
 
 	return nil
 }
+
+func restoreSnapshotData(ctx context.Context, homeDir string, clientName string) error {
+	log.Info(ctx, fmt.Sprintf("⌛ Restoring latest chain snapshot for %s...", clientName))
+
+	clientStorageDir, err := getClientStorageDir(clientName)
+	if err != nil {
+		return err
+	}
+
+	absClientStorageDir := filepath.Join(homeDir, clientStorageDir)
+	origAbsClientStorageDir := fmt.Sprintf("%s_orig_%d", absClientStorageDir, time.Now().Unix())
+
+	// Add _orig_<unix_timestamp> suffix to client storage dir to keep it as a backup.
+	log.Info(ctx, fmt.Sprintf("Renaming %s to %s to keep as backup...", absClientStorageDir, origAbsClientStorageDir))
+	if err := os.Rename(absClientStorageDir, origAbsClientStorageDir); err != nil {
+		return errors.Wrap(err, fmt.Sprintf("❌ Error renaming %s to %s to keep as backup", absClientStorageDir, origAbsClientStorageDir))
+	}
+
+	// Decompress backup snapshot archive file.
+	snapshotArchive := getSnapshotBackupArchive(clientName)
+	absSnapshotOutputDir := filepath.Join(homeDir, clientName)
+	if err := archive.DecompressTarLz4(ctx, snapshotArchive, absSnapshotOutputDir); err != nil {
+		return errors.Wrap(err, fmt.Sprintf("❌ Error restoring latest %s snapshot", clientName))
+	}
+
+	// Cleanup backup snapshot archive file.
+	if err := os.Remove(snapshotArchive); err != nil {
+		return errors.Wrap(err, "❌ Error removing latest snapshot backup archive: "+snapshotArchive)
+	}
+
+	switch clientName {
+	case gethClientName:
+		// Copy the newly generated jwtsecret file from the renamed geth home path to the actual geth home path with the newly restored snapshot data.
+		jwtSecretSrcFilePath := origAbsClientStorageDir + "/jwtsecret"
+		jwtSecretDestFilePath := absClientStorageDir + "/jwtsecret"
+		if err := fileio.Copy(jwtSecretSrcFilePath, jwtSecretDestFilePath); err != nil {
+			return errors.Wrap(err, fmt.Sprintf("❌ Error copying jwtsecret from %s to %s", origAbsClientStorageDir, absSnapshotOutputDir))
+		}
+
+		// Copy the newly generated nodekey file from the renamed geth home path to the actual geth home path with the newly restored snapshot data.
+		nodekeyFileSrcPath := origAbsClientStorageDir + "/nodekey"
+		nodekeyFileDestPath := absClientStorageDir + "/nodekey"
+		if err := fileio.Copy(nodekeyFileSrcPath, nodekeyFileDestPath); err != nil {
+			return errors.Wrap(err, fmt.Sprintf("❌ Error copying nodekey from %s to %s", origAbsClientStorageDir, absSnapshotOutputDir))
+		}
+	case haloClientName:
+		chainStatusSrcPath := filepath.Join(homeDir, "chain_status.json")
```

Comment:
what is the purpose of chain_status, doesn't seem like it used for anything?

Reply:
`chain-status.json` is part of the snapshot data and when decompressed we move it to `halo/data` where it reads info about the latest block hash. On a node with an empty state there is no such file, it gets created when state builds up from genesis.

Reply:
chain_status.json isn't used by halo, it isn't needed, no need to download it

Reply:
Done.